File: extension/src/sidepanel/types/matching.ts
export interface RadarDimension {
  name: string
  score: number
  candidate_value: string
  jd_requirement: string
}

export interface SkillGap {
  skill: string
  required: boolean
  candidate_has: boolean
  level: string
}

export interface MatchResult {
  overall_score: number
  summary: string
  radar_dimensions: RadarDimension[]
  matched_skills: string[]
  missing_skills: string[]
  strengths: string[]
  weaknesses: string[]
  recommendation: string
}

File: backend/app/models/matching.py
from pydantic import BaseModel


class RadarDimension(BaseModel):
    name: str
    score: float
    candidate_value: str
    jd_requirement: str


class MatchResult(BaseModel):
    overall_score: float
    summary: str
    radar_dimensions: list[RadarDimension]
    matched_skills: list[str]
    missing_skills: list[str]
    strengths: list[str]
    weaknesses: list[str]
    recommendation: str
